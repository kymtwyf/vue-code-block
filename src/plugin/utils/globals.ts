export const pluginName = 'v-code-block';
