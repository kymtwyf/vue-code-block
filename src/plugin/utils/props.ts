
export const AllProps = {
	browserWindow: false,
	code: '',
	codeBlockRadius: '0.5rem',
	copyButton: true,
	copyIcons: true,
	copyTab: true,
	copyFailedText: 'Copy failed!',
	copyText: 'Copy Code',
	copySuccessText: 'Copied!',
	floatingTabs: true,
	height: 'auto',
	highlightjs: false,
	indent: 2,
	label: '',
	lang: 'javascript',
	maxHeight: 'auto',
	persistentCopyButton: false,
	prismjs: false,
	prismPlugin: false,
	runTab: false,
	runText: 'Run',
	tabGap: '0.25rem',
	tabs: false,
	theme: 'neon-bunny',
};
