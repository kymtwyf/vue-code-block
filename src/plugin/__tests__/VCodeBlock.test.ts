import VCodeBlock from '../VCodeBlock.vue';
import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import { pluginName } from '../utils/globals';


describe('VCodeBlock Component', () => {

	it('should mount the component', () => {
		const wrapper = mount(VCodeBlock, {
			props: {
				prismjs: true,
			}
		});

		const returnedProps = wrapper.getComponent(VCodeBlock).props();

		expect(returnedProps).toMatchSnapshot();
	});

	it('should emit run event', () => {
		const wrapper = mount(VCodeBlock, {
			props: {
				prismjs: true,
				runTab: true,
				tabs: true,
			}
		});

		wrapper.find(`.${pluginName}--tab-run`).trigger('click');

		expect(wrapper.emitted()).toHaveProperty('run');
	});


	// -------------------------------------------------- Errors //
	const throwErrors = {
		bothSet: '[vue-code-block]: You cannot have both prismjs and highlightjs props set at the same time.',
		neitherSet: '[vue-code-block]: You must set either the prismjs or highlightjs props.',
	};

	it('should throw error if both prismjs and highlightjs props are true', () => {
		expect(() => {
			mount(VCodeBlock, {
				props: {
					prismjs: true,
					highlightjs: true,
				}
			});
		}).toThrowError(throwErrors.bothSet);
	});

	it('should throw error if both prismjs and highlightjs props are not set', () => {
		expect(() => {
			mount(VCodeBlock, {
				props: {
				}
			});
		}).toThrowError(throwErrors.neitherSet);
	});

	it('should throw error if neither prismjs or highlightjs props are true', () => {
		expect(() => {
			mount(VCodeBlock, {
				props: {
					prismjs: false,
					highlightjs: false,
				}
			});
		}).toThrowError(throwErrors.neitherSet);
	});

	it('should throw error if highlightjs used with prismPlugin prop', () => {
		expect(() => {
			mount(VCodeBlock, {
				props: {
					highlightjs: true,
					prismPlugin: true,
				}
			});
		}).toThrowError('[vue-code-block]: Highlight.js does not support PrismJS plugins. Unexpected results may occur. Remove the `prism-plugin` prop from the vue-code-block component.');
	});
});
