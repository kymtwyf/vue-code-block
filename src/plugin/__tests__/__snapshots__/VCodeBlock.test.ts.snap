// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VCodeBlock Component > should mount the component 1`] = `
{
  "browserWindow": false,
  "code": "",
  "codeBlockRadius": "0.5rem",
  "copyButton": true,
  "copyFailedText": "Copy failed!",
  "copyIcons": true,
  "copySuccessText": "Copied!",
  "copyTab": true,
  "copyText": "Copy Code",
  "cssPath": undefined,
  "floatingTabs": true,
  "globalOptions": false,
  "height": "auto",
  "highlightjs": false,
  "indent": 2,
  "label": "",
  "lang": "javascript",
  "languages": undefined,
  "maxHeight": "auto",
  "persistentCopyButton": false,
  "prismPlugin": false,
  "prismjs": true,
  "runTab": false,
  "runText": "Run",
  "tabGap": "0.25rem",
  "tabs": false,
  "theme": "neon-bunny",
}
`;
